File: ForBusiness/VisualStudio/OLD/NCDataOnline/02/NCDataOnlineService/NCDataOnlineService/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace NCDataOnlineService
{
    public static class Log
    {
        // Fields
        private static object key = new object();

        // Methods
        public static void Write(string str, string logfile)
        {
            lock (key)
            {
                try
                {
                    FileInfo info = new FileInfo(logfile);
                    if (info.Length > 1048576)
                    {
                        string contents = Util.ReadAllText(logfile, 60, 1000);
                        File.WriteAllText(logfile + ".old", contents);
                        File.WriteAllText(logfile, "");
                    }
                }
                catch (Exception)
                {
                }
                StreamWriter writer = new StreamWriter(File.Open(logfile, FileMode.Append), Encoding.Default);
                writer.WriteLine(DateTime.Now.ToString("u") + ": " + str);
                writer.Close();
            }
        }
    }


}

File: ForBusiness/NetBeans/NCDataOnlineService/src/ncdataonlineservice/Log.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ncdataonlineservice;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;

/**
 *
 * @author tml
 */
public class Log {
    public static void Write(String str, String logfile) throws IOException
    {
        File info = new File(logfile);
        if(info.exists()){
            if (info.length() > 1048576)
            {
                String contents = Util.ReadAllText(logfile);
                Util.WriteAllText(logfile + ".old", contents);
                Util.WriteAllText(logfile, "");
            }
        }
        
        try
        {
            DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
            Calendar cal = Calendar.getInstance();
            FileWriter fw = new FileWriter(logfile, true);
            fw.write(dateFormat.format(cal.getTime()) + " - " + str);
            fw.write(System.getProperty("line.separator"));
            fw.close();
        }
        catch(IOException ioe)
        {
            System.err.println("IOException: " + ioe.getMessage());
        }
    }
}
